src/status_im/common/enter_seed_phrase/view.cljs

```diff
@@ -139,16 +141,32 @@
                                      words-exceeded?       (i18n/label :t/seed-phrase-words-exceeded)
                                      error-in-words?       (i18n/label :t/seed-phrase-error)
                                      @invalid-seed-phrase? (i18n/label :t/seed-phrase-invalid)
-                                     :else                 (i18n/label :t/seed-phrase-info))]
+                                     :else                 (i18n/label :t/seed-phrase-info))
+          error-state?             (= suggestions-state :error)
+          button-disabled?         (or error-state?
+                                       (not (constants/seed-phrase-valid-length word-count))
+                                       (not all-words-valid?))]
       [:<>
-       [recovery-form
-        {:seed-phrase           @seed-phrase
-         :error-state?          (= suggestions-state :error)
-         :all-words-valid?      all-words-valid?
+       [recovery-phrase-form
+        {:title                 title
+         :seed-phrase           @seed-phrase
          :on-change-seed-phrase on-change-seed-phrase
          :word-count            word-count
-         :on-submit             on-submit
-         :keyboard-shown?       @keyboard-shown?}]
+         :on-input-ref          on-input-ref}
+        (if (fn? render-controls)
+          (render-controls {:submit-disabled?        button-disabled?
+                            :keyboard-shown?         @keyboard-shown?
+                            :container-style         (style/continue-button @keyboard-shown?)
+                            :prepare-seed-phrase     prepare-seed-phrase
+                            :focus-input             focus-input
+                            :seed-phrase             @seed-phrase
+                            :set-invalid-seed-phrase set-invalid-seed-phrase})
```

Comment:
Here we're passing in the required props to the `render-controls` function (if provided). Though I'm a little concerned about passing the raw `seed-phrase` around. Should we be masking the seed-phrase? Perhaps we don't need to mask the seed-phrase until we confirm it's a valid seed-phrase. Thoughts?

Reply:
Here's what some of the props are responsible for:
* `prepare-seed-phrase` - cleans and masks the seed-phrase, which is used before calling the seed-phrase validator.
* `focus-input` - used to automatically show the keyboard after an error has occurred.
* `set-invalid-seed-phrase` - used to set the error state to true for the recovery-phrase predictive keyboard component.

Reply:
> Here we're passing in the required props to the `render-controls` function (if provided). Though I'm a little concerned about passing the raw `seed-phrase` around. Should we be masking the seed-phrase? Perhaps we don't need to mask the seed-phrase until we confirm it's a valid seed-phrase. Thoughts?

@cammellos What are the guidelines for us to mask a seed phrase?

When I recently implemented the new keypair generation, I masked the seed phrase all the time since it was in re-frame, and only unmaked when it was used.

Reply:
it makes sense to mask it most of the times, just to avoid accidentally getting it printed in logs

Reply:
Okay sounds good, I'll make sure to mask the seed-phrase when passing it around to `render-controls` 👍